driver/src/main/java/org/neo4j/driver/internal/handlers/ResetResponseHandler.java

```diff
@@ -56,8 +56,11 @@ public final void onRecord( Value[] fields )
 
     private void resetCompleted( boolean success )
     {
-        messageDispatcher.unMuteAckFailure();
-        resetCompleted( completionFuture, success );
+        messageDispatcher.clearCurrentError();
+        if (completionFuture != null)
```

Comment:
Why this null check is added? Did we changed as old behaviour here? 
Why this future could be null and is it safe to not have an `else` here?